From 4bd33a9e15bf531fcb5a7c7b3ab0fb26efd40dc4 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 14 Oct 2013 23:46:46 +0200
Subject: [PATCH 241/277] MINOR: http: use conn_init() to reinitialize the
 server connection

It's safer and easier to proceed using this function which sets all
the required fields.
---
 src/proto_http.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/proto_http.c b/src/proto_http.c
index af96f41..61ce7ba 100644
--- a/src/proto_http.c
+++ b/src/proto_http.c
@@ -4292,12 +4292,10 @@ void http_end_txn_clean_session(struct session *s)
 	s->target = NULL;
 
 	/* reinitialize the connection to the server */
+	conn_init(s->req->cons->conn);
+
 	s->req->cons->state     = s->req->cons->prev_state = SI_ST_INI;
 	s->req->cons->end = NULL;
-	s->req->cons->conn->obj_type = OBJ_TYPE_CONN;
-	s->req->cons->conn->t.sock.fd = -1; /* just to help with debugging */
-	s->req->cons->conn->flags = CO_FL_NONE;
-	s->req->cons->conn->err_code = CO_ER_NONE;
 	s->req->cons->err_type  = SI_ET_NONE;
 	s->req->cons->conn_retries = 0;  /* used for logging too */
 	s->req->cons->exp       = TICK_ETERNITY;
-- 
1.8.1.5

